SimMuon/GEMDigitizer/src/GEMSimpleModel.cc

```diff
@@ -270,37 +271,31 @@ std::vector<std::pair<int, int> > GEMSimpleModel::simulateClustering(const GEMEt
 								     const PSimHit* simHit, const int bx, 
 								     CLHEP::HepRandomEngine* engine)
 {
-  const StripTopology& topology = roll->specificTopology(); // const LocalPoint& entry(simHit->entryPoint());
-  const LocalPoint& hit_position(simHit->localPosition());
-  const int nstrips(roll->nstrips());
-  int centralStrip = 0;
-  if (!(topology.channel(hit_position) + 1 > nstrips))
-    centralStrip = topology.channel(hit_position) + 1;
-  else
-    centralStrip = topology.channel(hit_position);
-  const GlobalPoint& pointSimHit = roll->toGlobal(hit_position);
-  const GlobalPoint& pointDigiHit = roll->toGlobal(roll->centreOfStrip(centralStrip));
-  double deltaX = pointSimHit.x() - pointDigiHit.x();
+  const LocalPoint& hit_entry(simHit->entryPoint());
+  const LocalPoint& hit_exit(simHit->exitPoint());
+
+  float hit_entry_smeardX;
+  float hit_exit_smeardX;
+  if (hit_entry.x()>hit_exit.x()) {
+    hit_entry_smeardX = hit_entry.x()+std::abs(CLHEP::RandGaussQ::shoot(0, resolutionX_));
```

Comment:
I think these random calls need to be: `CLHEP::RandGaussQ::shoot(engine, 0, resolutionX_)`
in order to use the specific instance of the random engine for this module.